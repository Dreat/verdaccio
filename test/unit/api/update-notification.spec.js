// @prettier
// @flow

import { 
    createBanner, 
    verdaccioUpdateNotification as notification 
} from '../../../src/lib/update-notification';

jest.mock('request', () => (url: string, resolver: Function) => {
    const response = {
        body: JSON.stringify({version: '4.5.6' })
    }
    resolver(null, response);
});

describe('Verdaccio update notification', () => {
    let log;

    beforeEach(() => {
        // mocking global console.log method
        global.console.log = jest.fn();
        log = global.console.log
    });

    test('should print major update notification', () => {
        notification('3.0.0');
        expect(log).toMatchSnapshot();
    });

    test('should print minor update notification', () => {
        notification('4.0.0');
        expect(log).toMatchSnapshot();
    });

    test('should print path update notification', () => {
        notification('4.5.0');
        expect(log).toMatchSnapshot();
    });

    test('when local version is equals to npm version', () => {
        notification('4.5.6');
        expect(log).not.toHaveBeenCalledWith();
    })

    test('when local version is greater than npm version', () => {
        notification('4.5.7');
        expect(log).not.toHaveBeenCalledWith();
    })
});

describe('create banner', () => {
    test('should create a banner', () => {
        expect(createBanner('1.0.0', '2.0.0', 'major')).toMatchSnapshot(); 
        expect(createBanner('1.0.0', '1.1.0', 'minor')).toMatchSnapshot(); 
        expect(createBanner('1.0.0', '1.0.1', 'patch')).toMatchSnapshot(); 
    })
});
