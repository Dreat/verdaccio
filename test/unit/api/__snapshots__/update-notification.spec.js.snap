// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verdaccio update notification should print major update notification 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "
        [37m[1mA new MAJOR version of Verdaccio is available on NPM. [1m[31m3.0.0[37m[1m â†’ [1m[32m4.5.6[37m[1m [22m[39m
        Run [32m[1mnpm install -g verdaccio[22m[39m to update.
        [34m[1mChangelog: https://github.com/verdaccio/verdaccio/releases/tag/v4.5.6[22m[39m
    ",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`Verdaccio update notification should print minor update notification 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "
        [37m[1mA new MINOR version of Verdaccio is available on NPM. [1m[31m4.0.0[37m[1m â†’ [1m[32m4.5.6[37m[1m [22m[39m
        Run [32m[1mnpm install -g verdaccio[22m[39m to update.
        [34m[1mChangelog: https://github.com/verdaccio/verdaccio/releases/tag/v4.5.6[22m[39m
    ",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`Verdaccio update notification should print path update notification 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "
        [37m[1mA new PATCH version of Verdaccio is available on NPM. [1m[31m4.5.0[37m[1m â†’ [1m[32m4.5.6[37m[1m [22m[39m
        Run [32m[1mnpm install -g verdaccio[22m[39m to update.
        [34m[1mChangelog: https://github.com/verdaccio/verdaccio/releases/tag/v4.5.6[22m[39m
    ",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`create banner should create a banner 1`] = `
"
        [37m[1mA new MAJOR version of Verdaccio is available on NPM. [1m[31m1.0.0[37m[1m â†’ [1m[32m2.0.0[37m[1m [22m[39m
        Run [32m[1mnpm install -g verdaccio[22m[39m to update.
        [34m[1mChangelog: https://github.com/verdaccio/verdaccio/releases/tag/v2.0.0[22m[39m
    "
`;

exports[`create banner should create a banner 2`] = `
"
        [37m[1mA new MINOR version of Verdaccio is available on NPM. [1m[31m1.0.0[37m[1m â†’ [1m[32m1.1.0[37m[1m [22m[39m
        Run [32m[1mnpm install -g verdaccio[22m[39m to update.
        [34m[1mChangelog: https://github.com/verdaccio/verdaccio/releases/tag/v1.1.0[22m[39m
    "
`;

exports[`create banner should create a banner 3`] = `
"
        [37m[1mA new PATCH version of Verdaccio is available on NPM. [1m[31m1.0.0[37m[1m â†’ [1m[32m1.0.1[37m[1m [22m[39m
        Run [32m[1mnpm install -g verdaccio[22m[39m to update.
        [34m[1mChangelog: https://github.com/verdaccio/verdaccio/releases/tag/v1.0.1[22m[39m
    "
`;
